src/components/activePanel/directives/activePanel.directive.js

```diff
@@ -15,61 +15,90 @@
                     scope: {},
                     link: function (scope, element) {
 
-                        var durationToDisplay,
-                            timerInterval,
-                            liveSessionData,
-                            liveSessionStatus = 0,
-                            liveSessionDuration = 0,
-                            timerSecondInterval = 1000,
-                            activePanelVisibleClassName = 'activePanel-visible',
-                            isStudent = ENV.appContext.toLowerCase() === 'student',
-                            isTeacher = ENV.appContext.toLowerCase() === 'dashboard',
-                            prevLiveSessionStatus = UserLiveSessionStateEnum.NONE.enum,
-                            bodyDomElem = angular.element($window.document.body),
-                            translateNamespace = 'ACTIVE_PANEL';
-
-                        $translate([
-                            translateNamespace + '.' + 'SHOW_STUDENT_SCREEN',
-                            translateNamespace + '.' + 'SHOW_TEACHER_SCREEN',
-                            translateNamespace + '.' + 'SHARE_MY_SCREEN',
-                            translateNamespace + '.' + 'END_SCREEN_SHARING'
-                        ]).then(function (translation) {
-                            scope.d.translatedStrings = {
-                                SHOW_STUDENT_SCREEN: translation[translateNamespace + '.' + 'SHOW_STUDENT_SCREEN'],
-                                SHOW_TEACHER_SCREEN: translation[translateNamespace + '.' + 'SHOW_TEACHER_SCREEN'],
-                                SHARE_MY_SCREEN: translation[translateNamespace + '.' + 'SHARE_MY_SCREEN'],
-                                END_SCREEN_SHARING: translation[translateNamespace + '.' + 'END_SCREEN_SHARING']
+                        const timerSecondInterval = 1000;
+                        const activePanelVisibleClassName = 'activePanel-visible';
+                        const isStudent = ENV.appContext.toLowerCase() === 'student';
+                        const isTeacher = ENV.appContext.toLowerCase() === 'dashboard';
+                        let durationToDisplay;
+                        let timerInterval;
+                        let liveSessionData;
+                        let liveSessionDuration = 0;
+                        let prevLiveSessionStatus = UserLiveSessionStateEnum.NONE.enum;
+
+
+                        activePanelInit();
+
+                        function activePanelInit() {
+                            getTranslations();
+
+                            scope.d = {
+                                states: {
+                                    NONE: 0,
+                                    LIVE_SESSION: 1
+                                },
```

Comment:
this should be outside of the function, no ?

Reply:
It should be on the view modal because I use it in the template